parsers/events.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+let addedCharacters = 0;
+
+function splitPreviousEvent(previousEvent, voicesAddedToEvent, eventIndex) {
+  const voicesToAdd = [];
+
+  let hasAddedCharacter = false;
+  // Split any previous events
+  previousEvent.forEach((voice) => {
+    if (voicesAddedToEvent.indexOf(voice.voice) === -1 &&
+        eventIndex < voice.index + voice.content.toString().length) {
+      // Split voice from previous event, add remainder to this event.
+      const splitIndex = eventIndex - voice.index + addedCharacters;
+      const event = {
+        index: eventIndex + addedCharacters,
+        voice: voice.voice,
+        content: `-${voice.content.substring(splitIndex)}`
+      };
+      voice.content = voice.content.substring(0, splitIndex);
+
+      voicesToAdd.push(event);
+
+      // Update index position to account for added character.
+      if (!hasAddedCharacter) {
+        addedCharacters++;
+        hasAddedCharacter = true;
+      }
+    }
+  });
+
+  return voicesToAdd;
+}
+
+function addEvents(verse, longestEvent) {
+  const voices = [];
+  const voicesAdded = [];
+
+  verse.forEach((events, voiceName) => {
+    if (events.length === 0) {
+      return;
```

Comment:
In what instance would the events of a phrase be empty?

Reply:
I've been treating the phrase map as a stack, where I pop off the top element of their event list. Once all events of a voice are processed, it ends as an empty stack.

It might make more sense to just delete that key from the map when a voice pops its last event.

An example where we process everything from `c1` before `v1`.

```
[ {
  c1: [ { index: 0, content: 'C' } ],
  v1: [ { index: 0, content: 'The' }, { index: 4, content: 'End' } ]
} ]
```